drivers/charger/charger_pf1550.c

```diff
@@ -0,0 +1,695 @@
+{
+	const struct charger_pf1550_config *const config = dev->config;
+	uint16_t idx;
+	uint8_t val;
+	int ret;
+
+	ret = linear_range_group_get_index(charger_fast_charge_ua_range, ARRAY_SIZE(charger_fast_charge_ua_range), current_ua, &idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val = FIELD_PREP(CHARGER_CHG_CURR_CFG, idx);
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      CHARGER_CHG_CURR_CFG,
+				      GENMASK(4,0),
+				      val);
+}
+
+static int pf1550_set_vbus_ilim(const struct device *dev, uint32_t current_ua)
+{
+	const struct charger_pf1550_config *const config = dev->config;
+	uint16_t idx;
+	uint8_t val;
+	int ret;
+
+	ret = linear_range_group_get_index(charger_vbus_ilim_range, ARRAY_SIZE(charger_vbus_ilim_range), current_ua, &idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val = FIELD_PREP(GENMASK(7,3), idx);
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      CHARGER_VBUS_INLIM_CNFG,
+				      GENMASK(7,3),
+				      val);
+}
+
+static int pf1550_set_vsys_min(const struct device *dev, uint32_t voltage_uv)
+{
+	const struct charger_pf1550_config *const config = dev->config;
+	uint16_t idx;
+	uint8_t val;
+	int ret;
+
+	ret = linear_range_group_get_index(charger_vsysmin_uv, ARRAY_SIZE(charger_vsysmin_uv), voltage_uv, &idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val = FIELD_PREP(GENMASK(7,6), idx);
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      CHARGER_BATT_REG,
+				      GENMASK(7,6),
+				      val);
+}
+
+static int pf1550_set_charge_termination_uv(const struct device *dev, uint32_t voltage_uv)
+{
+	const struct charger_pf1550_config *const config = dev->config;
+	uint16_t idx;
+	uint8_t val;
+	int ret;
+
+	ret = linear_range_group_get_index(charger_battery_termination_uv_range, ARRAY_SIZE(charger_battery_termination_uv_range), voltage_uv, &idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val = FIELD_PREP(GENMASK(5,0), idx);
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      CHARGER_BATT_REG,
+				      GENMASK(5,0),
+				      val);
+}
+
+static int pf1550_set_thermistor_mode(const struct device *dev,
+					enum charger_pf1550_therm_mode mode)
+{
+	const struct charger_pf1550_config *const config = dev->config;
+	uint8_t val = mode;
+	val = FIELD_PREP(GENMASK(1, 0), val);
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      CHARGER_THM_REG_CNFG,
+				      GENMASK(1, 0),
+				      val);
+}
+
+static int pf1550_set_enabled(const struct device *dev, bool enable)
+{
+	struct charger_pf1550_data *data = dev->data;
+	const struct charger_pf1550_config *const config = dev->config;
+
+	data->charger_enabled = enable;
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      CHARGER_CHG_OPER,
+				      GENMASK(1,0),
+				      enable ? 2 : 0);
+}
+
+static int pf1550_get_interrupt_source(const struct device *dev, uint8_t *int_a)
+{
+	const struct charger_pf1550_config *config = dev->config;
+	uint8_t dummy;
+	uint8_t *int_src;
+
+	int_src = (int_a != NULL) ? int_a : &dummy;
```

Comment:
I would flip this around, always read into dummy and then `if (int_a = NULL) {*int_a = dummy;}`